frontend/src/components/elements/ArrowVegaLiteChart/ArrowVegaLiteChart.tsx

```diff
@@ -428,13 +428,19 @@ function getDataArray(
   }
 
   const dataArr = []
-  const [rows, cols] =
-    dataProto.data.length > 0
-      ? [dataProto.data.length, dataProto.data[0].length]
-      : [0, 0]
-
-  const indexType = dataProto.types.index[0].name
-  const hasSupportedIndex = SUPPORTED_INDEX_TYPES.has(indexType)
+  const rows = dataProto.data.length
+  const cols = dataProto.data[0].length
+
+  // For `PeriodIndex` and `IntervalIndex` types are kept in `numpy_type`,
+  // for the rest of the indexes in `pandas_type`.
+  const {
+    pandas_type: pandasType,
+    numpy_type: numpyType,
+  } = dataProto.types.index[0]
+  const indexType = pandasType === "object" ? numpyType : pandasType
+  const hasSupportedIndex = SUPPORTED_INDEX_TYPES.has(
+    indexType as IndexTypeName
+  )
```

Comment:
Do we have tests for these different cases (numpy_type for PeriodIndex, IntervalIndex; pandas_type for other index types) for ArrowVegaLiteChart? I was looking in e2e and `ArrowVegaLiteChart.test.tsx` and didn't see them. Feels like a fiddly-enough implementation detail that we should enshrine it in a test. Is there a simple unit test you could create that just asserts `hasSupportedIndex` (or something) for various real-world scenarios?

Reply:
Addressed